diff --git a/tools/rkcommon.c b/tools/rkcommon.c
index af2a4835f6..5979a62569 100644
--- a/tools/rkcommon.c
+++ b/tools/rkcommon.c
@@ -78,7 +78,7 @@ static struct spl_info spl_infos[] = {
 	{ "rk3328", "RK32", 0x8000 - 0x1000, false },
 	{ "rk3368", "RK33", 0x8000 - 0x1000, false },
 	{ "rk3399", "RK33", 0x30000 - 0x2000, false },
-	{ "px30", "RK33", 0x2800, false },
+	{ "px30", "RK33", 0x40000, false },
 	{ "rv1108", "RK11", 0x1800, false },
 	{ "rk1808", "RK18", 0x200000 - 0x2000, false},
 };
